horace_core/sqw/file_io/@faccess_sqw_v3/put_bytes.m

```diff
@@ -0,0 +1,26 @@
+function obj = put_bytes(obj, to_write)
+%PUT_BYTES Write the given array to the sqw file as single
+%
+BYTES_IN_SINGLE = 4;
+DTYPE = 'float32';
+
+block_size = get(hor_config, 'mem_chunk_size');
+nbytes_to_write = BYTES_IN_SINGLE*numel(to_write);
+
+if nbytes_to_write <= block_size
```

Comment:
Does the `if` gain you much over using the loop generally? In the case of small loop counts it's one loop anyway.